tests/unit/specs/scripts/common.spec.js

```diff
@@ -25,3 +25,22 @@ describe(`coinsToObject`, () => {
     })
   })
 })
+
+describe(`toMicroDenom`, () => {
+  it(`returns the right micro denom of a given network`, () => {
+    const microDenom = toMicroDenom(`ATOM`)
+    expect(microDenom).toBe(`uatom`)
+  })
+})
+
+describe(`sleep`, () => {
+  it(`returns a true as a promise`, () => {
```

Comment:
hahahahahaha
this is not what this does
you should test if the promise resolves after a timeout

checkout jest.useFakeTimers and jest.runAllTimers

Reply:
Using jest.runAllTimers was kind of a headache. I think it is still testing the main functionality though

Reply:
the thing you want to test is: `it should resolve after x seconds`. then, inside the test, you use fake timers. now you run sleep with x seconds and store the promise. to later know that the promise resolved or not, you react to it resolving by setting a value when it does `sleep(x).then(() => done = true)`. now that the promise is initialized, we want to go to the future! there is a function `jest.forwardAllTimersBy(x)` or similar. We are now in the future. Now we can check if the promise was resolved by checking `done`

Reply:
@Bitcoinera see this comment?

Reply:
Oh, now I do :+1: missed this one